### PR TITLE
updated forge-sigchain for new merkle/path responses with hidden tree

### PR DESCRIPTION
**Pull request by AMarcedone**

### CODE REVIEW

**Comment by maxtaco** [comment]
does running this update change any of the older test cases? they should be deterministic

File: src/teamforge.iced [hunk 1]
```diff
@@ -548,7 +554,11 @@ class Team
         version: 2
         debug_payload: proof_gen_out.inner.obj
         debug_link_id: link_id
-        hidden_is_fresh : (if link_desc.hidden_is_fresh? then link_desc.hidden_is_fresh else true)
+        hidden_response: { 
+          resp_type: if @last_committed_hidden_tail == null then 2 else 3 # 2 denotes an absence proof, 3 an inclusion proof for a non nil leaf
```
Comment (maxtaco) [inline src/teamforge.iced]: use `not(foo?)` rather than `foo == null`

File: src/teamforge.iced [hunk 2]
```diff
@@ -801,18 +811,28 @@ class Team
...
+
+  #-------------------
+
+  _last_seqno : ->
     if @links.length > 0
-      @links[@links.length-1].for_client.seqno + 1
+      @links[@links.length-1].for_client.seqno 
     else
-      1
+      0
```
Comment (maxtaco) [inline src/teamforge.iced]: doesn't this change the previous compilation output?

Comment (AMarcedone) [reply src/teamforge.iced]: I don't think so, maybe I am missing something? 
Spot checking the output here https://github.com/keybase/keybase-test-vectors/pull/27/ also seems to be ok.

**Pull request merged**